macros/support/src/class/class.rs

```diff
@@ -217,6 +219,25 @@ impl Class {
             }
         }
     }
+
+    fn safe_query_interface(&self) -> TokenStream {
+        quote! {
+            pub fn query_interface<T: ::com::Interface>(self: &::std::pin::Pin<::std::boxed::Box<Self>>) -> Option<T> {
+                let mut result = None;
+                let hr = unsafe { self.QueryInterface(&T::IID, &mut result as *mut _ as _) };
+
+                if ::com::sys::FAILED(hr) {
+                    assert!(
+                        hr == ::com::sys::E_NOINTERFACE || hr == ::com::sys::E_POINTER,
```

Comment:
FYI. In theory, E_POINTER would only be returned if the second parameter is null. The caller can guarantee that this is not the case. In practice, implementations will typically AV by dereferencing the pointer without bothering to check since this is a coding error by the caller. Either way, asserting here is not advised as wonky implementations may return some non-standard result.

Reply:
Since only `E_NOINTERFACE` is expected it probably doesn't make a lot of sense to return `Err(hr)` instead?

Reply:
@MarijnS95 Hmm I'm not sure what errors would be recoverable from here? I think in general, it's just a question of whether it worked or not, but perhaps you're right that for ultimate flexibility we should be returning an HResult on error so that if some implementation returns a non-standard error, the user can inspect that...